core/templates/domain/platform_feature/feature-status-summary-object.factory.ts

```diff
@@ -45,9 +45,25 @@ export type FeatureSummaryDict = {
   }
 };
 
+/**
+ * Item of the summary of feature flags, which represents the status of one
+ * feature flag, providing the '.isEnabled' interface to check the status
+ * of that feature flag.
+ */
 class FeatureSummaryDictItem {
```

Comment:
Something about the name of this class is confusing to me. What about ``FeatureStatusChecker``? I don't like 'summary' since it's a highly overloaded term in the Oppia codebase, and this doesn't look like other summary objects that we have.

Reply:
In general, let's try to avoid 'summary' here & elsewhere if it's not entirely clear what's being summarized. Also, lets be consistent on 'platform feature' vs 'feature' vs 'feature flag' in this PR & subsequent PRs.

Reply:
Done, renamed it to `FeatureStatusChecker`, also `FeatureStatusCheckerItem`.
Also renamed the member in `PlatformFeatureService` from `featureSummary` to `status`.